sdk/program/src/message.rs

```diff
@@ -439,6 +439,12 @@ impl Message {
             accounts,
         })
     }
+
+    pub fn signers(&self) -> Vec<&Pubkey> {
+        self.account_keys[..self.header.num_required_signatures as usize]
```

Comment:
Could this panic on a badly constructed message? (if `header.num_required_signatures` > `account_keys.len()`)

Reply:
Yeah.  Do you think it makes more sense to return error or clamp the index to `max(num_required_signatures, account_keys.len())`

Reply:
I lean toward clamping the index, since `sanitize()` is available for more detail on construction failures. Could add a comment linking to that method.